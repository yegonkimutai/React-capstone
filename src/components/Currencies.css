.main-container {
  font-family: 'Franklin Gothic Medium', 'Arial Narrow', Arial, sans-serif;
}

.home-title {
  font-size: 2.5rem;
  text-align: center;
}

.search-input {
  width: 90%;
  margin-left: 1rem;
  height: 30px;
}

.currency-list {
  display: grid;
  grid-template-columns: 1fr 1fr;
  text-align: center;
  margin-top: 1rem;
}

.currency-card {
  background-color: #81ce7c;
  border: none;
}

.currency-card:nth-child(4n+2),
.currency-card:nth-child(4n+3) {
  background-color: #eddb34;
}
