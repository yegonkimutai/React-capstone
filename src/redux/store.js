import { configureStore } from "@reduxjs/toolkit";
import seasonReducer from "./seasons/seasonSlice";

const store = configureStore({
    reducer: {
        seasons: seasonReducer
    }
})

export default store