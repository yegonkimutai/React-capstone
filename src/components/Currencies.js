import { useState } from 'react';
import { useSelector } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { FaChevronDown, FaChevronUp } from 'react-icons/fa';
import './Currencies.css';

const Currencies = () => {
  const navigation = useNavigate();
  const { currencies } = useSelector((store) => store.currencies);
  const [search, setSearch] = useState('');

  const filtered = currencies.filter((currency) => currency.symbol
    .toLowerCase()
    .includes(search.toLowerCase()));

  const switchPage = (currency) => {
    navigation(`information/${currency.name}`, { state: { currency } });
  };

  return (
    <div className="main-container">
      <h2 className="home-title">Currency Tracker</h2>
      <div>
        <input
          type="text"
          placeholder="Search currency..."
          className="search-input"
          value={search}
          onChange={(e) => setSearch(e.target.value)}
          role="button"
          tabIndex={0}
        />
      </div>
      <div className="currency-list">
        {filtered.map((currency) => (
          <button
            type="button"
            key={currency.id}
            onClick={() => { switchPage(currency); }}
            onKeyDown={(e) => {
              if (e.target === 'Enter') {
                switchPage(currency);
              }
            }}
            className="currency-card"
          >
            <h2>{currency.symbol}</h2>
            <p className="change">
              {currency.percent_change_1h < 0 ? (
                <>
                  <FaChevronDown style={{ color: 'red' }} />
                  <span style={{ color: 'red' }}>
                    {Math.abs(currency.percent_change_1h)}
                    %
                  </span>
                </>
              ) : (
                <>
                  <FaChevronUp style={{ color: 'green' }} />
                  <span style={{ color: 'green' }}>
                    {currency.percent_change_1h}
                    %
                  </span>
                </>
              )}
            </p>
          </button>
        ))}
      </div>
    </div>
  );
};

export default Currencies;
