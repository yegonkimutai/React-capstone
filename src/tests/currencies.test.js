import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import Currencies from "../components/Currencies";
import store from "../redux/store";
import TestRenderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';

describe('Currencies', () => {
    test('Render Currencies Component', () => {
        const tree = TestRenderer.create(
            <BrowserRouter>
            <Provider store={store}>
            <Currencies />
            </Provider>
            </BrowserRouter>
        ).toJSON()
        expect(tree).toMatchSnapshot()
    })
})