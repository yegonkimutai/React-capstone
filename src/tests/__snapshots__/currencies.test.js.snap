// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Currencies Render Currencies Component 1`] = `
<div
  className="main-container"
>
  <h2
    className="home-title"
  >
    Currency Tracker
  </h2>
  <div>
    <input
      className="search-input"
      onChange={[Function]}
      placeholder="Search currency..."
      role="button"
      tabIndex={0}
      type="text"
      value=""
    />
  </div>
  <div
    className="currency-list"
  />
</div>
`;
